examples/grpc/experimental/greeter_interceptor.cc

```diff
@@ -0,0 +1,199 @@
+#include "messages.grpc.pb.h"
+#include "tracer_common.h"
+
+#include <grpcpp/grpcpp.h>
+#include <grpcpp/server.h>
+#include <grpcpp/server_builder.h>
+#include <grpcpp/server_context.h>
+
+#include <chrono>
+#include <fstream>
+#include <string>
+#include <thread>
+
+#include <grpcpp/channel.h>
+#include <grpcpp/client_context.h>
+#include <grpcpp/create_channel.h>
+#include <grpcpp/generic/generic_stub.h>
+#include <grpcpp/impl/codegen/proto_utils.h>
+#include <grpcpp/support/server_interceptor.h>
+
+using grpc::Server;
+using grpc::ServerBuilder;
+using grpc::ServerContext;
+using grpc::ServerWriter;
+using grpc::Status;
+
+using grpc_example::Greeter;
+using grpc_example::GreetRequest;
+using grpc_example::GreetResponse;
+
+class GreeterInterceptor : public grpc::experimental::Interceptor
+{
+public:
+  GreeterInterceptor(grpc::experimental::ServerRpcInfo *info)
+  {
+    info_                                        = info;
+    const char *method                           = info->method();
+    grpc::experimental::ServerRpcInfo::Type type = info->type();
+  }
+
+  void Intercept(grpc::experimental::InterceptorBatchMethods *methods) override
+  {
+    if (methods->QueryInterceptionHookPoint(
+            grpc::experimental::InterceptionHookPoints::PRE_SEND_INITIAL_METADATA))
+    {
+      auto *map = methods->GetSendInitialMetadata();
```

Comment:
The real use case of interceptors for instrumentation would be to intercept the requests, and create spans out of the request attributes to be exported to external endpoint. While it is ok for now to have `cout` here, please update the README for this as todo/comming soon.